src/tokenizer/index.js

```diff
@@ -40,14 +44,28 @@ function codePointToString(code) {
 }
 
 export default class Tokenizer {
-  constructor(options, input) {
+  // Forward-declarations
+  // location.js
+  +raise: (pos: number, message: string) => empty;
+  // comments.js (TODO: Better type for the parameter)
+  +addComment: (comment: Object) => void;
+  // parser/index.js
+  +hasPlugin: (name: string) => boolean;
+  // parser/util.js
+  +unexpected: (pos?: ?number, messageOrType?: string | TokenType) => empty;
```

Comment:
I wonder if these methods are in the wrong place currently when they are used but assign later, especially `raise` and `unexpected` and maybe other helper functions could be moved either here or into a new baseclass.
But this is out of scope for this PR.